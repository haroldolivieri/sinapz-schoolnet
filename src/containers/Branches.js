import React from 'react'
import { BranchList } from '../components'

class Branches extends React.Component {
  render() {
    return (
      <div>
        <h1>Filiais</h1>
        <BranchList branches={this.props.branches} />
      </div>
    )
  }
}

export default Branches