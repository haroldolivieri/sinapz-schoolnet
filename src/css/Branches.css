.branch-list-container {
  display: flex;
  flex-direction: column;
  flex: 1;
}