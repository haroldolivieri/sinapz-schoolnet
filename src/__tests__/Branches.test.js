import React from 'react'
import Branches from '../containers/Branches'
import { shallow } from 'enzyme'

describe(Branches, () => {

  context('with branch list', () => {
    let branches = [{ id: 1, name: 'madureira', inCharge: 'Hadoldo' }, { id: 2, name: 'Meier', inCharge: 'Paula' }]
    const wrapper = shallow(<Branches branches={branches} />)

    it('renders correctlyt', () => { expect(wrapper).toMatchSnapshot() })
  })

  context('with empty branch list', () => {
    const wrapper = shallow(<Branches branches={[]} />)
    
    it('should match with snapshot', () => { expect(wrapper).toMatchSnapshot() })
  })
})
