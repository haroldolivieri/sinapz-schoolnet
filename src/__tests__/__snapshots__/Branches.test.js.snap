// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branches with branch list renders correctlyt 1`] = `
<div>
  <h1>
    Filiais
  </h1>
  <BranchList
    branches={
      Array [
        Object {
          "id": 1,
          "inCharge": "Hadoldo",
          "name": "madureira",
        },
        Object {
          "id": 2,
          "inCharge": "Paula",
          "name": "Meier",
        },
      ]
    }
  />
</div>
`;

exports[`Branches with empty branch list should match with snapshot 1`] = `
<div>
  <h1>
    Filiais
  </h1>
  <BranchList
    branches={Array []}
  />
</div>
`;
